File: src/dataset/cpp/sieve_exercism-felbit/claude/ClaudeSonnet4_sieve_v1.cpp
#include "sieve.h"

namespace sieve {
    auto primes(int n) -> const std::vector<int>
    {
        if (n < 2) return {};
        
        std::vector<bool> is_prime(n + 1, true);
        std::vector<int> result;
        
        is_prime[0] = is_prime[1] = false;
        
        for (int i = 2; i * i <= n; ++i) {
            if (is_prime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    is_prime[j] = false;
                }
            }
        }
        
        result.reserve(n / (std::log(n) > 0 ? std::log(n) : 1));
        
        for (int i = 2; i <= n; ++i) {
            if (is_prime[i]) {
                result.push_back(i);
            }
        }
        
        return result;
    }
}

File: src/dataset/javascript/sieve_exercism-elenamountz/claude/ClaudeSonnet4_sieve_v1.js
export default class Sieve{
  constructor(number){
    this.number = number;
    this.list = Array.from(Array(this.number - 1), (value, index) => index+2);
    this.primes = this.findPrimes();
  }
  findPrimes(){
    const isPrime = new Array(this.number + 1).fill(true);
    isPrime[0] = isPrime[1] = false;

    for(let i = 2; i * i <= this.number; i++){
      if(isPrime[i]){
        for(let j = i * i; j <= this.number; j += i){
          isPrime[j] = false;
        }
      }
    }

    const primes = [];
    for(let i = 2; i <= this.number; i++){
      if(isPrime[i]){
        primes.push(i);
      }
    }
    return primes;
  }
}